lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -1362,21 +1361,14 @@ checkRewardIsWorthTxCost
 checkRewardIsWorthTxCost txWitnessTag pp balance delegationTemplateM = do
     when (balance == Coin 0)
         $ Left ErrWithdrawalNotBeneficial
-    let minimumCost txCtx =
-            calculateMinimumFee feePerByte txWitnessTag txCtx emptySkeleton
-        costWith = minimumCost $ mkTxCtx balance
-        costWithout = minimumCost $ mkTxCtx $ Coin 0
-        worthOfWithdrawal = Coin.toInteger costWith - Coin.toInteger costWithout
-    when (Coin.toInteger balance < 2 * worthOfWithdrawal)
+    let costOfWithdrawal = _txRewardWithdrawalCost feePerByte witType balance
+    when (Coin.toInteger balance < 2 * Coin.toInteger costOfWithdrawal)
```

Comment:
@Anviking: can you remind me why we use `2 *` here?

Perhaps it'd be worth adding a comment to describe the justification. (Though can be in a later PR.)

Reply:
In the end opted not to add one as I felt I'd mostly be repeating the function name.

But if `withdrawalAmt == costOfWithdrawal`, then we're completely wasting the reward balance. Waiting for more rewards would be better. Here we ensure we can at most burn half of the withdrawal amount on withdrawal-induced fees.